collect_app/src/main/java/org/odk/collect/android/provider/FormsProvider.java

```diff
@@ -325,21 +324,20 @@ public int delete(@NonNull Uri uri, String where, String[] whereArgs) {
                         if (c != null && c.getCount() > 0) {
                             c.moveToFirst();
                             do {
-                                deleteFileOrDir(c.getString(c
-                                        .getColumnIndex(FormsColumns.JRCACHE_FILE_PATH)));
-                                String formFilePath = c.getString(c
-                                        .getColumnIndex(FormsColumns.FORM_FILE_PATH));
-                                deleteFileOrDir(formFilePath);
-                                deleteFileOrDir(c.getString(c
-                                        .getColumnIndex(FormsColumns.FORM_MEDIA_PATH)));
+                                deleteFileOrDir(storageManager.getAbsoluteCacheFilePath(c.getString(c
+                                        .getColumnIndex(FormsColumns.JRCACHE_FILE_PATH))));
+                                deleteFileOrDir(storageManager.getAbsoluteFormFilePath(c.getString(c
+                                        .getColumnIndex(FormsColumns.FORM_FILE_PATH))));
+                                deleteFileOrDir(storageManager.getAbsoluteFormFilePath(c.getString(c
+                                        .getColumnIndex(FormsColumns.FORM_MEDIA_PATH))));
 
                                 try {
                                     // get rid of the old tables
                                     ItemsetDbAdapter ida = new ItemsetDbAdapter();
                                     ida.open();
-                                    ida.delete(c.getString(c
+                                    ida.delete(storageManager.getRelativeFormFilePath(c.getString(c
```

Comment:
This is going to do a database query, right? So it seems like it should match whatever is in the database? I'd expect no transformation needed here.

Reply:
Right, fixed.